File: python/319_Bulb_Switcher.py
import math


class Solution(object):
    def bulbSwitch(self, n):
        """
        :type n: int
        :rtype: int
        """
        return int(math.sqrt(n))


def test():
    s = Solution()
    print s.bulbSwitch(5)
    # final state: 1 0 0 1 0 , n = 2


if __name__ == '__main__':
    test()

File: C++/319_Bulb_Switcher.cpp
#include <iostream>

using namespace std;

class Solution {
public:
	int bulbSwitch(int n) {
		return (int)sqrt(n);
	}
};

int main(){
	Solution s;
	cout << s.bulbSwitch(5);
	// final state: 1 0 0 1 0, n = 2
	return 0;
}